.azure-pipelines/bazel.yml

```diff
@@ -17,6 +17,11 @@ parameters:
     default: true
 
 steps:
+  - bash: |
+      /usr/local/bin/set-instance-protection.sh on
```

Comment:
can we do some generic hook script in the image, so we don't have to update job?

Reply:
The problem is there are many things running outside of the image (like cache steps) that wouldn't be covered if we started using the images.

Reply:
If you mean in the AMI that we build no, again the jobs offer no hooks into when a job has started/stopped so there's no accurate way of setting them

Reply:
well I mean in AMI, just a name, instead of `set-instance-protection.sh`, say `on-job-start.sh`/`on-job-finish.sh`.

Reply:
I mean we can do that, but IMO we should protect the instances first, using what was already merged so they stop being manually managed for now :)

Reply:
OK, though this shouldn't be hard-coded locally, the azp yml are not auto merged so everyone have to merge master.